cap-primitives/src/rustix/linux/fs/open_impl.rs

```diff
@@ -61,23 +69,6 @@ pub(crate) fn open_beneath(
         Mode::empty()
     };
 
-    // On Android, seccomp kills processes that execute unrecognized system
-    // calls, so we do an explicit version check rather than relying on
-    // getting an `ENOSYS`.
-    #[cfg(target_os = "android")]
-    {
-        static CHECKED: AtomicBool = AtomicBool::new(false);
-
-        if !CHECKED.load(Relaxed) {
-            if !openat2_supported() {
-                INVALID.store(true, Relaxed);
-                return Err(rustix::io::Errno::NOSYS.into());
-            }
-
-            CHECKED.store(true, Relaxed);
-        }
-    }
```

Comment:
AFACIT removing this is wrong, you need to always return `NONSYS` instead.

That's because there are other callers of this function, e.g.: https://github.com/bytecodealliance/cap-std/blob/1317c28a93e8ce9d8b66d4319e705290e9a4167c/cap-primitives/src/rustix/linux/fs/stat_impl.rs#L11-L23

Reply:
```rust
pub(crate) fn open_beneath(
    start: &fs::File,
    path: &Path,
    options: &OpenOptions,
) -> io::Result<fs::File> {
    #[cfg(target_os = "android")]
    {
        return Err(rustix::io::Errno::NOSYS.into());
    }
```

This fix definitely works (tested on an actual Android impl that hit both `openat2` and `statx` issues before now - an Oculus Quest 2, which makes this an extremely side-tracked thing for me so I will very likely not be opening a PR myself).

Reply:
Thanks for the report! Confirmed that this what's happening. I've now filed https://github.com/bytecodealliance/cap-std/pull/316 to fix this, and disable `openat2` on Android entirely with a `cfg`.